pyccel/codegen/printing/ccode.py

```diff
@@ -1688,7 +1691,18 @@ def _print_Allocate(self, expr):
         free_code = ''
         variable = expr.variable
         if isinstance(variable.class_type, (HomogeneousListType, HomogeneousSetType, DictType)):
-            return ''
+            if expr.status in ('allocated', 'unknown'):
```

Comment:
Your PR description makes it sound like the unknown case is not handled

Reply:
Updated the description